### PR TITLE
fix(vue): watch for scope dispose

### PR DESCRIPTION
**Pull request by harlan-zw**
Reactivity outside of components seems to be broken (see https://github.com/danielroe/nuxt-turnstile/issues/66#issuecomment-1332827264).

This commit (https://github.com/unjs/unhead/commit/92e17d34908cb185cff5e3d5e0a0e32f509bce8c) should fix that, but it may lead to issues where the watchers will never get disposed properly.

We update the internal logic to use scope effects to allow users to opt-out.

### CODE REVIEW

**Comment by harlan-zw** [comment]
Hey @danielroe 

Bit of a noob with effectScope, would you mind reviewing this code :pray: 

Seems like it fixes the reactivity issue you flagged and tests are passing, but don't want to introduce any new issues.

File: examples/nuxt3/plugins/script.ts [hunk 1]
```diff
@@ -21,14 +22,24 @@ const script = {
...
   const addScript = ref(false)
 
-  useHead({
-    script: () => [
-      { children: configure },
-      addScript.value && script,
-    ].filter((s): s is typeof script => !!s),
+  let scope : EffectScope | null = effectScope()
+
+  scope.run(() => {
```
Comment (harlan-zw) [inline examples/nuxt3/plugins/script.ts:27]: just an example, I'd imagine most users wouldn't do this so the watchers will never be stopped?

**Comment by danielroe** [comment]
this looks like the right approach to me - I think you're right to be cautious of reactivity outside a vue component. but given we're on client-side, shouldn't any watchers setup in a plugin persist for the entire lifecycle of the nuxt app?

**Comment by harlan-zw** [comment]
Thanks! Yes the watchers from the `useHead` method will persist for the entire lifecycle unless wrapped in an effectScope and disposed of manually

Have dropped wrapping internal watchers with useEffect

**Comment by harlan-zw** [comment]
Might actually avoid this for now just to minimise any new errrors. Will re-implement if there's an actual requirement for it from someone.

**Pull request closed without merging**